kubectl-cloudflow/cmd/verify.go

```diff
@@ -0,0 +1,170 @@
+	cmd *cobra.Command
+}
+
+func init() {
+	verifyBlueprintCMD := &verifyBlueprintCMD{}
+	verifyBlueprintCMD.cmd = &cobra.Command{
+		Use:     "verify",
+		Short:   "Verifies the blueprint of a Cloudflow Application.",
+		Example: "kubectl cloudflow verify <blueprint-file-name>",
+		Run:     verifyBlueprintCMD.verifyImpl,
+		Args:    validateVerifyCmdArgs,
+	}
+	rootCmd.AddCommand(verifyBlueprintCMD.cmd)
+}
+
+// this map is constructed entirely from blueprint
+func getImageRefsFromConfig(config *configuration.Config) map[string]domain.ImageReference {
+	// get the images from the blueprint
+	images := config.GetNode("blueprint.images").GetObject().Items()
+
+	var imageKeyVals = make(map[string]domain.ImageReference)
+
+	for imageKey, imageRef := range images {
+		ref, err := verify.ParseImageReference(imageRef.GetString())
+		if err != nil {
+			util.LogAndExit(err.Error())
+		}
+		imageKeyVals[imageKey] = *ref
+	}
+	return imageKeyVals
+}
+
+func getFilePath(pathToBlueprint string) (string, error) {
+	blueprintURL, err := url.Parse(pathToBlueprint)
+	if blueprintURL == nil || err != nil {
+		return "", fmt.Errorf("You need to specify the full path to a blueprint file. %s, is malformed. Error: %s", pathToBlueprint, err.Error())
+	}
+	if blueprintURL.Scheme == "" { // this is a local file
+		return pathToBlueprint, nil
+	}
+
+	// url
+	timeout := time.Duration(1 * time.Second)
+	client := http.Client{
+		Timeout: timeout,
+	}
+
+	tempDir, err := getTempDir()
+
+	if err != nil {
+		return "", fmt.Errorf("Local temp dir not properly defined. Error: %s", err.Error())
+	}
+	resp, err := client.Get(pathToBlueprint)
+
+	if err != nil {
+		return "", fmt.Errorf("You need to specify the full path to a blueprint file. %s, is not accessible. Error: %s ", pathToBlueprint, err.Error())
+	}
+
+	if resp == nil || resp.StatusCode != 200 {
+		return "", fmt.Errorf("You need to specify the full path to a blueprint file. %s, is not accessible", pathToBlueprint)
+	}
+
+	file, err := ioutil.TempFile(tempDir, path.Base(blueprintURL.Path)+".")
+	if err != nil {
+		return "", fmt.Errorf("Could not create a temp file. Error: %s", err.Error())
+	}
+
+	if _, err = io.Copy(file, resp.Body); err != nil {
+		return "", fmt.Errorf("Could not copy the remote blueprint file.  Error: %s", err.Error())
+	}
+
+	if err = file.Close(); err != nil {
+		return "", fmt.Errorf("Failed to close the file.  Error: %s", err.Error())
+	}
+	return file.Name(), nil
+}
+
+func (c *verifyBlueprintCMD) verifyImpl(cmd *cobra.Command, args []string) {
+	// check blueprint contents here
+	blueprint := args[0]
+
+	file, err := getFilePath(blueprint)
+	if err != nil {
+		util.LogAndExit("Failed to fetch blueprint contents from %s", blueprint)
+	}
+
+	content, err := loadBlueprintFile(file)
+	err = verify.VerifyBlueprint(content)
+	if err != nil {
+		util.LogAndExit("Blueprint verification failed. Error: %s", err.Error())
+	} else {
+		print("Blueprint verification succeeded.")
```

Comment:
We have a set of utility functions that can be used to print when an operation succeeds, `util.PrintSuccess`, with fancy ansi coloring :)